opslevel/resource_opslevel_system.go

```diff
@@ -1,162 +1,224 @@
+			"note": schema.StringAttribute{
+				Description: "Additional information about the system.",
+				Optional:    true,
+			},
+			"owner": schema.StringAttribute{
+				Description: "The id of the team that owns the system.",
+				Optional:    true,
+				Validators:  []validator.String{IdStringValidator()},
+			},
+		},
+	}
+}
+
+func (r *SystemResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel SystemResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	systemInput := opslevel.SystemInput{
+		Name:        planModel.Name.ValueStringPointer(),
+		Description: planModel.Description.ValueStringPointer(),
+		OwnerId:     opslevel.NewID(planModel.Owner.ValueString()),
+		Note:        planModel.Note.ValueStringPointer(),
+	}
+	if planModel.Domain.ValueString() != "" {
+		systemInput.Parent = opslevel.NewIdentifier(planModel.Domain.ValueString())
+	}
+	system, err := r.client.CreateSystem(systemInput)
+	if err != nil || system == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create system, got error: %s", err))
+		return
+	}
+	stateModel, diags := NewSystemResourceModel(ctx, *system)
+	if diags != nil && diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	stateModel.LastUpdated = timeLastUpdated()
+
+	tflog.Trace(ctx, "created a system resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *SystemResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var planModel SystemResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readSystem, err := r.client.GetSystem(planModel.Id.ValueString())
+	if err != nil || readSystem == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to read system, got error: %s", err))
+		return
+	}
+	stateModel, diags := NewSystemResourceModel(ctx, *readSystem)
+	if diags != nil && diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *SystemResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var planModel SystemResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	systemInput := opslevel.SystemInput{
+		Name:        planModel.Name.ValueStringPointer(),
+		Description: planModel.Description.ValueStringPointer(),
+		OwnerId:     opslevel.NewID(planModel.Owner.ValueString()),
+		Note:        planModel.Note.ValueStringPointer(),
+	}
+	if planModel.Domain.IsNull() {
```

Comment:
I think lines 185-189 can just be removed, and inside the block on 179 you can put

```
		Parent:        opslevel.NewIdentifier(planModel.Domain.ValueString())
```

If the domain is null in config it would evaluate to `""` which would become `null` upon JSON marshal.